goal_src/jak1/engine/ui/hud.gc

```diff
@@ -270,6 +308,10 @@
   :code (behavior ()
     (logior! (-> self mask) (process-mask sleep-code))
     (loop
+      ;; if all collectable hud parts are hidden, revert collectables view to original
+      (if (and (!= *hud-collectable-view* (hud-collectable-view original)) (hud-collectables-hidden?))
+        (set! *hud-collectable-view* (hud-collectable-view original))
+        )
```

Comment:
Add some sort of notice that you changed this from the original game.

The counters resetting when they go away feels like only a half-way implementation of this. It's probably way more intuitive and easier to mod if you just change the `hud-update` methods for each of the hud classes separately instead of doing it all in a state which *all* hud classes use. It also probably gives you more control.

Reply:
Sure, I can move this into just the fuel/money/buzzer `hud-update` methods, that makes sense though it will be a bit of repeated code.

I can remove the reset on when the hud is hidden, that's easy - but I do want to leave the state global so that all 3 hud parts are in sync (i.e. avoid situations like cells showing game-totals while buzzer shows level-totals)